### PR TITLE
Clean old interfaces

### PR DESCRIPTION
**Pull request by PetrovKP**
- Remove old interfaces (interface1, interface2, ...)
- Remove inner folders for DAAL interfaces

Reduce size

|                                    |  Before (MB)   | After (MB) | Diff   |
|------------------------ |----------------|-------------|------|
| [LNX] onedal_core.a   | 1230              | 969            | 260   |
| [LNX] onedal_core.so  | 231               | 200            | 31     |
| [WIN] onedal_core.lib  | 840              | 640            | 200   |
| [WIN] onedal_core.dll  | 101               | 91             | 10     |

### CODE REVIEW

**Comment by PetrovKP** [comment]
/intelci: start
set infra_branch=delete_prev_ifaces

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/ebce3f7f-08b3-f125-89fa-8cdcd4b723a1

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/ebce4987-c263-f1a1-a170-8cdcd4b723a1

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/ebceab94-3877-f1c2-8ceb-8cdcd4b723a1

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces
compatibility_branch=2020-clean-iface
compatibility_repo=https://github.com/intel-innersource/libraries.performance.data-analytics.dal-infra.git
compatibility_revision=HEAD

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces
compatibility_branch=2020-clean-iface
compatibility_repo=https://github.com/intel-innersource/libraries.performance.data-analytics.dal-infra.git
compatibility_revision=HEAD

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces
compatibility_branch=2020-clean-iface
compatibility_repo=https://github.com/intel-innersource/libraries.performance.data-analytics.dal-infra.git
compatibility_revision=HEAD

**Comment by PetrovKP** [comment]
/intelci: run
set infra_branch=delete_prev_ifaces
compatibility_branch=2020-clean-iface
compatibility_repo=https://github.com/intel-innersource/libraries.performance.data-analytics.dal-infra.git
compatibility_revision=HEAD

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/ebd1ca75-79af-f160-a7ce-8cdcd4b723a1

File: cpp/daal/src/algorithms/logitboost/logitboost_train_friedman_aux.i [hunk 1]
```diff
@@ -38,7 +38,7 @@
 #ifndef __LOGITBOOST_TRAIN_FRIEDMAN_AUX_I__
 #define __LOGITBOOST_TRAIN_FRIEDMAN_AUX_I__
 
-#include "src/algorithms/stump/inner/stump_train_impl.i"
+// #include "src/algorithms/stump/inner/stump_train_impl.i"
```
Comment (owerbat) [inline cpp/daal/src/algorithms/logitboost/logitboost_train_friedman_aux.i:41]: Delete?

**Review by owerbat — APPROVED** [review]

**Comment by PetrovKP** [comment]
@Mergifyio rebase

**Comment by PetrovKP** [comment]
http://intel-ci.intel.com/ebd85623-8e09-f130-bb89-8cdcd4b723a1

**Pull request merged**